### PR TITLE
Add provision to `visit_BlockCall` in "pass_utils" and `visit_Block` in "asr_to_llvm".

### LINKED ISSUE #1262: ASR Verify Failed: Function "xx"  depends on "yy" but isn't found in dependency list.
This will be reproduced after getting #1261 merged. I have fix ready for it, will open up a PR once it gets merged.
For
```fortran
subroutine hybrd()
    implicit none
    real, dimension(3):: wa3
    real :: temp

    interface 
    subroutine fcn(n)
    implicit none
    integer, intent(in) :: n
    end subroutine fcn
    end interface

    main : block
        temp = enorm(3, Wa3)
    end block main

end subroutine
```

> LFortran

```console
$ lfortran ./tests/temp.f90 --implicit-typing --implicit-interface --show-llvm
ASR verify pass error: ASR verify: Function hybrd depends on enorm but isn't found in its dependency list.
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/pranavchiku/lfortran/src/bin/lfortran", in _start()
  File "./csu/../csu/libc-start.c", line 392, in __libc_start_main_impl()
  File "./csu/../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main()
  File "/home/pranavchiku/lfortran/src/bin/lfortran.cpp", line 1862, in ??
    return emit_llvm(arg_file, lfortran_pass_manager,
  File "/home/pranavchiku/lfortran/src/bin/lfortran.cpp", line 775, in ??
    = fe.get_llvm(input, lm, pass_manager, diagnostics);
  File "/home/pranavchiku/lfortran/src/lfortran/fortran_evaluator.cpp", line 287, in LCompilers::FortranEvaluator::get_llvm(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::LocationManager&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&)
    Result<std::unique_ptr<LLVMModule>> res = get_llvm2(code, lm, pass_manager, diagnostics);
  File "/home/pranavchiku/lfortran/src/lfortran/fortran_evaluator.cpp", line 309, in LCompilers::FortranEvaluator::get_llvm2(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::LocationManager&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&)
    diagnostics, lm.files.back().in_filename);
  File "/home/pranavchiku/lfortran/src/lfortran/fortran_evaluator.cpp", line 342, in LCompilers::FortranEvaluator::get_llvm3(LCompilers::ASR::TranslationUnit_t&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    compiler_options, run_fn, infile);
  File "/home/pranavchiku/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 6820, in LCompilers::asr_to_llvm(LCompilers::ASR::TranslationUnit_t&, LCompilers::diag::Diagnostics&, llvm::LLVMContext&, Allocator&, LCompilers::PassManager&, LCompilers::CompilerOptions&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    pass_manager.apply_passes(al, &asr, pass_options, diagnostics);
  File "/home/pranavchiku/lfortran/src/libasr/pass/pass_manager.h", line 209, in LCompilers::PassManager::apply_passes(Allocator&, LCompilers::ASR::TranslationUnit_t*, LCompilers::PassOptions&, LCompilers::diag::Diagnostics&)
    _apply_passes(al, asr, _passes, pass_options, diagnostics);
LCompilersException: Verify failed
````

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #1262, via commit b8d0aee0f360fc7bf92440b8ac68d20e15115cac, which then results in "Function code not generated for `enorm`" which is then fixed by 0dd407b93f900e1518c411c22e83dec37ece736a.

### CODE REVIEW

File: tests/tests.toml [hunk 1]
```diff
@@ -2823,3 +2823,7 @@ asr = true
 [[test]]
 filename = "block2.f90"
 asr = true
+
+[[test]]
+filename = "block3.f90"
+llvm = true
```
Comment (certik) [inline tests/tests.toml:2829]: I would just move this into integration tests and run it. That's the best way to test the LLVM backend, unless there is some reason why that cannot be done.

**Comment by certik** [comment]
Otherwise it looks good to me. @czgdp1807 does it look good to you as well?

File: src/libasr/pass/pass_utils.h [hunk 2]
```diff
@@ -341,6 +341,17 @@ namespace LCompilers {
                     }
                     BaseWalkVisitor<UpdateDependenciesVisitor>::visit_SubroutineCall(x);
                 }
+
+                void visit_BlockCall(const ASR::BlockCall_t& x) {
+                    if( fill_function_dependencies ) {
```
Comment (czgdp1807) [inline src/libasr/pass/pass_utils.h:346]: Why should we not visit `Block::m_body` when `fill_function_dependencies` is `false`? Any specific reason?

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 3]
```diff
@@ -4383,6 +4386,20 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
             this->visit_stmt(*(associate_block->m_body[i]));
         }
     }
+    void visit_Block(const ASR::Block_t& x) {
```
Comment (czgdp1807) [inline src/libasr/codegen/asr_to_llvm.cpp:4389]:
```suggestion

    void visit_Block(const ASR::Block_t& x) {
```

File: src/libasr/pass/pass_utils.h (hunk 2 above)
Comment (Pranavchiku) [reply src/libasr/pass/pass_utils.h:346]: I tried to mimic `visit_FunctionCall`, and I guess we need no check for it. Will make changes.

File: tests/tests.toml (hunk 1 above)
Comment (czgdp1807) [reply tests/tests.toml:2829]: @Pranavchiku This is yet to be addressed. Please remove this test from here and also delete the associated reference files (they are still showing up in the diff).

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: integration_tests/block_04.f90 [hunk 4]
```diff
@@ -0,0 +1,30 @@
+subroutine hybrd()
+    implicit none
+    real, dimension(3):: wa3
+    real :: temp
+
+    interface 
+    subroutine fcn(n)
+    implicit none
+    integer, intent(in) :: n
+    end subroutine fcn
```
Comment (certik) [inline integration_tests/block_04.f90:10]:
```suggestion
        subroutine fcn(n)
        implicit none
        integer, intent(in) :: n
        end subroutine fcn
```

**Comment by Pranavchiku** [comment]
I am not sure why these references are getting updated, it happened after I did `git rebase main` to resolve conflicts.

**Comment by Thirumalai-Shaktivel** [comment]
Did you make a clean build (`git clean -dfx && ./build1.sh`)?

**Comment by Pranavchiku** [comment]
Yeah, I just saw a small mistake, I made while resolving conflicts.

**Comment by Thirumalai-Shaktivel** [comment]
Do this, `git reset main`, stage changes, and make new commits.
Follow this steps:
https://github.com/lcompilers/lpython/blob/8fa533a43860f43ce10e6dee238147fad57ac205/doc/src/rebasing.md#merging

**Comment by Thirumalai-Shaktivel** [comment]
If you get stuck somewhere or face the same issue, let me know.

**Comment by Pranavchiku** [comment]
Yeah sure!

**Comment by Pranavchiku** [comment]
Closing with reference to #1284.

**Pull request closed without merging**